### PR TITLE
[Copy] Updates strings on Search page

### LINKED ISSUE #6991: Copy changes for Search page
### Working language ability

- [ ] Please add: 

The selected working language ability will be compared to the one chosen by candidates in their applications. To note, candidates who selected Bilingual may not have Government of Canada second language evaluation results.

La capacité linguistique sélectionnée sera comparée à celle choisie par les candidats dans leur demande. À noter, les candidats qui ont sélectionné Bilingue peuvent ne pas avoir des résultats d’évaluation de langue seconde du gouvernement du Canada.

### Work location

- [ ] Please add: 

You can select more than one region.

Vous pouvez sélectionner plus d’une région.

### Text for no matches: 

We didn't find any matching candidates.
We can still help! 

- [ ] Please change second line to:

We may be able help! 
Nous pouvons peut-être vous aider!

### Filtre de classification
For IT-04, it currently says: Conseiller principal ou Manager Gestionnaire
- [ ] Please remove “Manager”  (just keep “Gestionnaire”)

### PR DESCRIPTION
**Pull request by mnigh**
🤖 Resolves #6991.

## 👋 Introduction

This PR changes English and French copy on the Search page.

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. `npm run build`
2. Navigate to `/search`
3. Observe changes to English and French copy based on [linked issue](https://github.com/GCTC-NTGC/gc-digital-talent/issues/6991)

## 📸 Screenshots
Screen Shot 2023-06-19 at 11 56 02
Screen Shot 2023-06-19 at 11 56 40
Screen Shot 2023-06-21 at 10 08 19
Screen Shot 2023-06-21 at 10 08 34
Screen Shot 2023-06-19 at 12 00 08
Screen Shot 2023-06-19 at 12 00 41
Screen Shot 2023-06-19 at 12 01 12
Screen Shot 2023-06-19 at 12 05 28

### CODE REVIEW

File: apps/web/src/pages/SearchRequests/SearchPage/components/SearchForm.tsx [hunk 1]
```diff
@@ -502,8 +502,8 @@ const SearchForm = React.forwardRef<SearchFormRef, SearchFormProps>(
             })}
             text={intl.formatMessage({
               defaultMessage:
-                "If you have more detailed work location requirement, let us know in the comment section of the submission form.",
-              id: "v7sYE7",
+                "If you have more detailed work location requirement, let us know in the comment section of the submission form. You can select more than one region. If you are open to all regions, select All Regions.",
+              id: "fJ35xz",
```
Comment (esizer) [inline apps/web/src/pages/SearchRequests/SearchPage/components/SearchForm.tsx:506]: I recognize this is the copy provided but, it is confusing since there is no "All Regions" option. At first galnce (especially because of the uppercase letters) I was expecting "All Regions" to be an option in the list. Maybe we make this lower case or reword it to be clearer?

"If you are open to all regions, select all available regions"? 🤔

**Review by esizer — CHANGES_REQUESTED** [review]
I have some concerns about the clarity regarding work regions. Other than that, this looks great!

File: apps/web/src/pages/SearchRequests/SearchPage/components/SearchForm.tsx (hunk 1 above)
Comment (mnigh) [reply apps/web/src/pages/SearchRequests/SearchPage/components/SearchForm.tsx:506]: Good catch. @gobyrne can you weigh in on this suggestion and if you approve of it, provide its French equivalent? I'm not sure if a @GCTC-NTGC/designers also would need to update a design somewhere (there wasn't a linked figma in the issue).

Comment (gobyrne) [reply apps/web/src/pages/SearchRequests/SearchPage/components/SearchForm.tsx:506]: Passing the question along to recruitment team

**Comment by gobyrne** [comment]
@mnigh @esizer ok, so recruitment team actually wants to add a new option for "All Regions". I'll work with @Jerryescandon to break this off into a new issue. We'll need to define matching logic at a minimum. Can you merge the rest of it?

File: apps/web/src/pages/SearchRequests/SearchPage/components/SearchForm.tsx (hunk 1 above)
Comment (mnigh) [reply apps/web/src/pages/SearchRequests/SearchPage/components/SearchForm.tsx]: @esizer I have removed the problematic copy **If you are open to all regions, select All Regions.**, please see df3c7e4616602f1cd790ca3cf7c5e63613ae3c77.

**Review by esizer — APPROVED** [review]
LGTM! 👍
